Access Refactor v2

HalfdeadKiller opened the pull request:
## About the Pull Request

A very large edit of ID references, and job IDs. Almost every SS13 job has been commented out. Site 19 jobs have been moved from the maps folder to the code/game/jobs/ folder. Need to solve confusion with jobs being defined in two places though, with site53 taking priority when loading.

access_datum.dm has comments regarding access levels in general, and what I've cross-referenced with existing jobs.

I've done testing on the ID Computer, as I've edited stuff in card.dm, _access_defs.dm, access.dm. Everything seems to work, but there's no test like production.

PLEASE GO OVER MY CHANGES TO JOB ACCESS, AND SUGGEST CHANGES IF THINGS DON'T MAKE SENSE. THESE ACCESS LEVELS ARE NOT SET IN STONE AND I ENCOURAGE READJUSTMENT BASED ON GROUP CONSENSUS.

MAPS MAY NEED TO BE UPDATED. I WILL TRY TO UPDATE SOME THINGS IN MAPS AS I FIND THEM IN ANOTHER PR, IF OTHERS DO NOT GET TO IT.

UPDATE ON MAPS: I used VSC to check for all SS13 strings in maps, VSC didn't find any. Here is the RegEx I used.
```
"ACCESS_SECURITY"|"ACCESS_BRIG"|"ACCESS_ARMORY"|"ACCESS_FORENSICS"|"ACCESS_MEDICAL"|"ACCESS_MORGUE"|"ACCESS_TOXINS"|"ACCESS_TOX_STORAGE"|"ACCESS_ENGINEERING"|"ACCESS_ENGINE_EQUIP"|"ACCESS_MAINT"|"ACCESS_EXTERNAL"|"ACCESS_EMERGENCY_STORAGE"|"ACCESS_CHANGE_ID"|"ACCESS_AI_UPLOAD"|"ACCESS_TELEPORTER"|"ACCESS_EVA"|"ACCESS_BRIDGE"|"ACCESS_CAPTAIN"|"ACCESS_PERSONAL_LOCKERS"|"ACCESS_CHAPEL_STORAGE"|"ACCESS_TECH_STORAGE"|"ACCESS_ATMOS"|"ACCESS_BAR"|"ACCESS_JANITOR"|"ACCESS_CREMATORIUM"|"ACCESS_KITCHEN"|"ACCESS_ROBOTICS"|"ACCESS_RESEARCH_DIRECTOR"|"ACCESS_CARGO"|"ACCESS_CONSTRUCTION"|"ACCESS_CHEMISTRY"|"ACCESS_CARGO_BOT"|"ACCESS_HYDROPONICS"|"ACCESS_MANUFACTURING"|"ACCESS_LIBRARY"|"ACCESS_LAWYER"|"ACCESS_VIRO"|"ACCESS_CHIEF_MEDICAL_OFFICER"|"ACCESS_QUARTERMASTER"|"ACCESS_NETWORK"|"ACCESS_SURGERY"|"ACCESS_RESEARCH"|"ACCESS_MINING"|"ACCESS_MINING_OFFICE"|"ACCESS_SORTING"|"ACCESS_VAULT"|"ACCESS_MINING_EVA"|"ACCESS_XENOBIO"|"ACCESS_CHIEF_ENGINEER"|"ACCESS_HEAD_OF_PERSONNEL"|"ACCESS_HEAD_OF_SECURITY"|"ACCESS_REQUEST_ANNOUCE"|"ACCESS_GATEWAY"|"ACCESS_SEC_DOORS"|"ACCESS_PSYCHIATRIST"|"ACCESS_XENOARCH"|"ACCESS_MEDICAL_EQUIP"|"ACCESS_HEADS"
```

## Why It's Good For The Game

Attempts to refactor and unify access levels in some way that makes sense.

## Changelog

:cl:
balance: Access Refactor v2
/:cl:

HalfdeadKiller committed:
Security Refactor

Replaces security access with security levels.

HalfdeadKiller committed:
Medical Refactor

Medical Access Refactor

HalfdeadKiller committed:
Science Refactor

Science Related Access

HalfdeadKiller committed:
Engineering Refactor

Engineering ID changes.

HalfdeadKiller committed:
Cargo Refactor

Edits cargo access stuff.

HalfdeadKiller committed:
Command Refactor

Attempt at Command ID editing. May need refinement.

HalfdeadKiller committed:
Refactord Command and Staff

Cleaned up access_datum.dm. Organized remaining station roles to the top.

HalfdeadKiller committed:
Merge branch 'AccessRefactorV2' of https://github.com/HalfdeadKiller/HDK-Foundation-19 into AccessRefactorV2

HalfdeadKiller committed:
Command, Logistics Refactor, ID Computer

THIS NEEDS CHECKING OVER OF CHANGES card.dm, _access_defs.dm, access.dm

HalfdeadKiller committed:
Job ID Cleanup

Attempts to make all access make sense. May need tweaking, since everything kind of intertwines.

HalfdeadKiller committed:
Merge branch 'Foundation-19:dev' into AccessRefactorV2

HalfdeadKiller committed:
Update command.dm

Missing Comma in List.

HalfdeadKiller committed:
Adds change_ids to Site Director

Tsurupeta commented on code/game/machinery/alarm.dm:
```diff
@@ -56,7 +56,7 @@
 	active_power_usage = 1000 //For heating/cooling rooms. 1000 joules equates to about 1 degree every 2 seconds for a single tile of air.
 	power_channel = ENVIRON
-	req_access = list(list(access_securitylvl1, access_atmospherics, access_engine_equip)) //access_securitylvl1 added since mapper air alarms have it configured.
+	req_access = list(list(access_securitylvl1, access_engineeringlvl2, access_engineeringlvl2)) //access_securitylvl1 added since mapper air alarms have it configured.
```
```suggestion
	req_access = list(list(access_securitylvl1, access_engineeringlvl2)) //access_securitylvl1 added since mapper air alarms have it configured.
```

HalfdeadKiller replied:
Shit. I forgot that some items had multiple access, and when replacing it ended up putting the same one twice. Original access sheet didn't have kitchen defined, and comments said it was so the cooks could have access to D-Class stuff for training purposes. There isn't really an access level for civilian jobs, and in an effort to keep it not cluttered, I used existing D-Class access instead.

HalfdeadKiller replied:
Fixed in 995546f

Tsurupeta commented on code/game/machinery/alarm.dm:
```diff
@@ -127,7 +127,7 @@
 /obj/machinery/alarm/server/New()
 	..()
-	req_access = list(access_rd, access_atmospherics, access_engine_equip)
+	req_access = list(access_sciencelvl5, access_engineeringlvl2, access_engineeringlvl2)
```
```suggestion
	req_access = list(access_sciencelvl5, access_engineeringlvl2)
```

HalfdeadKiller replied:
Fixed in 995546f

Tsurupeta commented on code/game/machinery/deployable.dm:
```diff
@@ -6,47 +6,47 @@ Barricades
+	access_adminlvl4 = 21
 	access_chapel_office = 22
-	access_tech_storage = 23
-	access_atmospherics = 24
-	access_bar = 25
-	access_janitor = 26
+	access_engineeringlvl2 = 23
+	access_engineeringlvl2 = 24
+	access_dclasskitchen = 25
+	access_dclassjanitorial = 26
 	access_crematorium = 27
-	access_kitchen = 28
-	access_robotics = 29
-	access_rd = 30
-	access_cargo = 31
-	access_construction = 32
-	access_chemistry = 33
-	access_cargo_bot = 34
-	access_hydroponics = 35
+	access_dclasskitchen = 28
+	access_sciencelvl2 = 29
+	access_sciencelvl5 = 30
+	access_logistics = 31
+	access_engineeringlvl2 = 32
+	access_medicallvl2 = 33
+	access_logistics = 34
+	access_dclassbotany = 35
 	access_manufacturing = 36
 	access_library = 37
-	access_lawyer = 38
-	access_virology = 39
-	access_cmo = 40
-	access_qm = 41
+	access_adminlvl1 = 38
+	access_medicallvl4 = 39
+	access_medicallvl5 = 40
+	access_adminlvl2 = 41
 	access_court = 42
 	access_clown = 43
 	access_mime = 44
```
wtf is this, did you even read your changes or blindly CTRL+F replace everything and put out PR?

HalfdeadKiller replied:
I read through everything it touched, decided if it was important or not, and then replaced all. I'd wager 80% of what has changed could actually be deleted, but I don't know if you all want a bunch of stuff deleted. That all appeared to be legacy ID stuff, commented out so people assigning access had an easier time adding every ID number together. None of it appears to be used.

Tsurupeta replied:
![image](https://user-images.githubusercontent.com/41485301/182021510-2ea53b3e-06ba-4c4f-a35b-9406cda96c20.png)
`I've read everything` moment

HalfdeadKiller replied:
Would you rather I just delete all of the commented out stuff I replaced, which is what that stuff is in your picture?

Tsurupeta commented on code/game/machinery/doors/firedoor.dm:
```diff
@@ -13,7 +13,7 @@
 	var/welded_file = 'icons/obj/doors/hazard/welded.dmi'
 	icon_state = "open"
-	req_access = list(list(access_atmospherics, access_engine_equip))
+	req_access = list(list(access_engineeringlvl2, access_engineeringlvl2))
```
```suggestion
	req_access = list(list(access_engineeringlvl2))
```

HalfdeadKiller replied:
Fixed in 995546f

Tsurupeta commented on code/game/machinery/kitchen/gibber.dm:
```diff
@@ -6,7 +6,7 @@
 	density = TRUE
 	anchored = TRUE
-	req_access = list(access_kitchen,access_morgue)
+	req_access = list(access_dclasskitchen,access_medicallvl2)
```
```suggestion
	req_access = list(access_dclasskitchen, access_medicallvl2)
```

Tsurupeta replied:
though I dont remember if it requires BOTH of these accesses or only one, I think you need to reassess it.

Tsurupeta commented on code/game/machinery/kitchen/smartfridge.dm:
```diff
@@ -74,7 +74,7 @@
 	desc = "A refrigerated storage unit for storing medicine and chemicals."
 	icon_contents = "chem"
-	req_access = list(list(access_medical,access_chemistry))
+	req_access = list(list(access_medicallvl2,access_medicallvl2))
```
```suggestion
	req_access = list(list(access_medicallvl2))
```

Tsurupeta reviewed (changes requested):
Looked at 1/3 of it, and have problem with you assigning `DCLASS_KITCHEN` instead of regular kitchen for stuff. D-class shit should be completely separate from regular kitchens and stuff.

Gona review more later.

HalfdeadKiller committed:
Update code/game/machinery/kitchen/smartfridge.dm

Co-authored-by: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>

HalfdeadKiller committed:
Merge branch 'Foundation-19:dev' into AccessRefactorV2

HalfdeadKiller commented:
"Space Station 13 areas.dm" needs access reviewed since I don't understand mapping code enough to understand the ramifications of areas having req_access = (%access%) defined.

HalfdeadKiller committed:
Removes some duplicate access definitions.

RegEx comparison to find obvious duplicated. There may still be some lurking around.

HalfdeadKiller committed:
Fixes build errors in crew_record.dm

Failed to check that I was removing duplicates in crew_record.dm, which requires a view access level, and a edit access level.

HalfdeadKiller commented:
Investigate Accounts_DB.dm access requirements. Current code appears to want Adminlvl5 or access_finances. Originally I think it was access_captain and access_hop. This may want to be redefined to be only access_finances, but may require code logic changes.

HalfdeadKiller committed:
Fixes some duplicate assignments.

HalfdeadKiller committed:
Update code/game/machinery/kitchen/gibber.dm

Co-authored-by: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>

HalfdeadKiller commented:
access_maint_tunnels needs some form of discussion on, since I'm not sure if mappers want to add it to maintenance doors or not. Some jobs I added it to, but it may need to be removed from some, depending on how access to doors is setup.

HalfdeadKiller committed:
Merge branch 'dev' into AccessRefactorV2

HalfdeadKiller committed:
Fix Merge Conflict, Duplicate Access

scientist.dm duplicate access fix.
medical.dm merge conflict resolved.

HalfdeadKiller committed:
Merge branch 'dev' into AccessRefactorV2

HalfdeadKiller committed:
Removes Virologist Job Datum

Removes Virologist Job Datum.

HalfdeadKiller commented:
I need to go over the changes in code/game/jobs/job/ since I think I got confused and edited those when other ones needed to be edited instead. Those changes may be unnecessary, or can be commented out.

HalfdeadKiller committed:
Duplicate Access in vending.dm

Duplicate Access in vending.dm

HalfdeadKiller committed:
Revert changed to cart.dm

cart.dm local variables got changed, and I failed to notice. This reverts the file back to it's original form.

HalfdeadKiller committed:
Fix mob Robot duplicate access

Fixes mob robot duplicate access. Also deletes virlogist from site53 medical.dm, instead of comment blocking the job out.

HalfdeadKiller committed:
Delete debug .NETFramework CS file

Likely was not supposed to have been commited in the first place.

HalfdeadKiller committed:
Delete CreditsTool.csproj cache

Wasn't supposed to be commited.

HalfdeadKiller committed:
Merge branch 'Foundation-19:dev' into AccessRefactorV2

HalfdeadKiller committed:
Merge branch 'Foundation-19:dev' into AccessRefactorV2

MixMashy reviewed (commented):
Mistake of a review. Sorry. Cannot delete

hry-gh commented:
hey @halfdeadkiller, are you planning on deconflicting this pr? no worries if not, just give us a shout

ivanmixo commented:
If this isn't alive I'm murdering it in a couple days

hry-gh commented:
rip... thank you for the effort making the pr, though
